static/js/biomedical/bio_charts_utils.ts

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as d3 from "d3";
+import _ from "lodash";
+
+import { GraphNodes } from "../shared/types";
+import { DiseaseGeneAssociationData } from "./disease/chart";
+import { InteractionLink, ProteinNode, ProteinNumData } from "./protein/chart";
+import { DiseaseAssociationType } from "./protein/page";
+export const MARGIN = { top: 30, right: 30, bottom: 90, left: 160 };
+// number to select top data points for large data
+export const NUM_DATA_POINTS = 10;
+// tooltip constant for all charts
+export const TOOL_TIP = d3
+  .select("#main")
+  .append("div")
+  .attr("class", "tooltip");
+// shift in tooltip positions
+export const TOOL_TIP_SHIFT = 60;
+// default brightness for barcharts
+export const DEFAULT_BRIGHTEN_PERCENTAGE = "112%";
+// x axis label shift
+export const X_LABEL_SHIFT = 40;
+export type Datum =
+  | DiseaseGeneAssociationData
+  | ProteinNumData
+  | ProteinNode
+  | InteractionLink
+  | DiseaseAssociationType;
+/**
+ * When mouse first enters element specified by given id, brighten it and update/display the global tooltip.
+ */
+export function onMouseOver(
+  elementID: string,
+  toolTipText: string,
+  brightenPercentage: string = DEFAULT_BRIGHTEN_PERCENTAGE
+): void {
+  // brighten element: https://stackoverflow.com/a/69610045
+  d3.select(`#${elementID}`).style(
+    "filter",
+    `brightness(${brightenPercentage})`
+  );
+  // update tooltip text
+  TOOL_TIP.html(toolTipText);
+  // show tooltip
+  TOOL_TIP.style("opacity", 1);
+}
+/**
```

Comment:
please add an empty line between functions

Reply:
Done! Although, I thought the running the linter should also fix issues like adding extra lines, perhaps not in this case:(